Add fyta to latest

muffin142 opened the pull request:
Please add my adapter ioBroker.fyta to latest.

*This pull request was created by https://www.iobroker.dev 9bea956.*

Commit:
Added fyta to sources-dist.json

mcm1957 commented:
reminder 10.2.2025

mcm1957 commented:
@muffin142 

Please provide object dump - see note above:

> To verify the object structure of this adapter during REVIEW please export the object structure of a working installation and attach the file to this PR. You find a guide how to export the object struture here: https://github.com/ioBroker/ioBroker.repochecker/blob/master/OBJECTDUMP.md

muffin142 commented:
Please find object dump attached

[fyta.0.json](https://github.com/user-attachments/files/18717882/fyta.0.json)

mcm1957 commented:
@muffin142 

First of all - THANK YOU for the time and effort you spend to maintain this adapter.

I would like to give some feedback based on my personal oppinion. @Apollon77 might have additional suggestions or even a different oppinion to one or the other statement. Please feel free to contact him if you cannot follow my suggestions or want to discuss some special aspects.

- [ ] Readme.md should contain a link to the manufacturer and/or device

  The README.md of any adapter related to device or m ultiple devices of a manufacturer should contain a link the manufacturers website and/or to a description of the device. This is to anable new users to easily check wether they associate the adapter with the correct device(s). (See https://github.com/ioBroker/ioBroker.repositories#requirements-for-adapter-to-get-added-to-the-latest-repository paragraph 4.)

- [ ] unused onStateChange handler

  You configred an onStateChange handler but it looks like this handler does not do any important tasks. Please remove the handler if the adapter does not react on state changes.

- [ ] Object type hirarchie

  Please note that only ONE object of type 'device' is allowed within the hirarchie. The structure must be
  [device] [- channel ..] - state. 'Device' may be omitted but if present channel must be below device. 'Channel' can be used multiple times. 'States' must not have any children. In addition objects of type 'folder' can be present at any level.

- [ ] structure of jsonConfig is invalid

  i18n is not an object containing translations but a single attribute with boolean avlue (true/false).
  if i18n is true translation data is placed in files located at i18n subtree. You only have to fill in the english version (i18b/en/translations.json and execute `npm run translate` to populate other languages.
  If i18n is set to false (NOT recommended) you may insert the translaations directly inot jsonConfig.json as multilanguage objects just like you did at io.package.json. But note that translation into all supported ioBroker languages is mandatore. You can use https://translator-ui.iobroker.in/ to generate the objects. But using i18b external files is recommended as this enables the usage ob weblate to gain support from external human translators too.

- [ ] jsonCostum.json5 seems to be invalid

  I did not detect handling of custom attributes by your adapter. If I'm correct and did not miss something, please remove jsonCostum.json5. And please remove  common.supprtsCustom from io-package.json too in this case (https://github.com/muffin142/ioBroker.fyta/blob/704eb0f0b6e315e81f4549c693b10308a58547b1/io-package.json#L77)

- [ ] outdated words.js

  words.js seems to be outdated and is empty. JsonConfig does not use this file at all. So please remove the file.

 - [ ] i18n entries must support all languages
 
  When using multilangugae support at objects those entries must contain all supported languages. See https://github.com/muffin142/ioBroker.fyta/blob/704eb0f0b6e315e81f4549c693b10308a58547b1/io-package.json#L140. You can use https://translator-ui.iobroker.in/ to generate i18n blocks.

- [ ] Add timeout to axios calls

  All axios calls should have a timeout set. The default value for axios timeout is 0 which results in no timeout. So an unresponsive remote system could lead to an hanging adapter. Either set an appropiate timeout per axios call or set a default timeout globally for the axios instance.
  
- [ ] consider using adapter.setTimeout / this.setTimeout instead of (standard) setTimeout
  
  The adapter package provides wrapper routines for native setTimeout, setInterval, clearTimeout and clearInterval. Using those routines ensures that timers are cancelled on on load. Additional checks on thomse limits might be performed, too. So consider replacing native setTimeout/clearTimeout by adapter.setTimeout/adapter.clearTimeout or this.setTimeout/this.clearTimeout. The same refers to setInterval/clearInterval.

- [ ] reenable testing
  
  Standard iobroker testing is required but is disabled. Please activate standard ./github/workflows/test-and-release.yml again. Of course its ok to add addional tests but standard github based tests must be active and named as provided by default to enable semi automatic checks. 
    
- [ ] linter setup

  Please use standard iobroker linter setup if acceptable. This is located at @iobroker/eslint-config. See migration guide at https://github.com/ioBroker/ioBroker.eslint-config/blob/main/MIGRATION.md.
    
- [ ] deleting of objects

  Not sure if deleting all objects is really a good idea as this will invalidate all custom configuration (i.e. history settings). But as it's an option set by the user this functionality is acceptable. BUT your code will delte info objects for any instance other than instance 0. This must be fixed. See https://github.com/muffin142/ioBroker.fyta/blob/704eb0f0b6e315e81f4549c693b10308a58547b1/main.js#L47

- [ ] error handling

  - [ ] Stopping processing must be logged as error (and not as info) - see https://github.com/muffin142/ioBroker.fyta/blob/704eb0f0b6e315e81f4549c693b10308a58547b1/main.js#L100 and 
https://github.com/muffin142/ioBroker.fyta/blob/704eb0f0b6e315e81f4549c693b10308a58547b1/main.js#L105
  
  - [ ] Adapter should try to recover from errors at least at limited number of times as accessing an external source may lead to an error. As the interval is set to 30 minutes anyway inlimited retryes would be acceptable too as logging an error every 30 minutes does not overlaod ioBroker for sure. 
    Simply stopping processing becaue i.e. the home router has an interruption due do to ip address change is not userfriendly. Same if the external webpage is ofline for a short time. But IF the adapter stops processing it must terminate itself so that user has a clear indicatio n taht it is no longer working. Simply doing nothing will be interpreted as an error of the adapter by most users.

Thanks for reading and evaluating this suggestions.
McM1957

**Please add a comment when you have reviewed and fixed the suggestionsor at least commented the suggestions and you think the adapter is ready for a re-review!**

**PLease add a new object dump after fixing type hirarchy**

muffin142 commented:
Hey,
I've adjusted the points and hopefully haven't overlooked anything. The fact that deleting objects can destroy the history is a fair point, I've added this to my backlog for a future version, as well as i18n support via individual language files.

Please also find the updated object dump attached.
[fyta.json](https://github.com/user-attachments/files/18788749/fyta.json)

As the updated package needs to be available on npm now there is a published version 0.1.3.

mcm1957 commented:
RE-CHECK!

mcm1957 commented:
WORK IN PROGRESS - This is a review checklist only - IGNORE FOR NOW
----------------

- [ ] Readme.md should contain a link to the manufacturer and/or device

  The README.md of any adapter related to device or m ultiple devices of a manufacturer should contain a link the manufacturers website and/or to a description of the device. This is to anable new users to easily check wether they associate the adapter with the correct device(s). (See https://github.com/ioBroker/ioBroker.repositories#requirements-for-adapter-to-get-added-to-the-latest-repository paragraph 4.)

- [ ] unused onStateChange handler

  You configred an onStateChange handler but it looks like this handler does not do any important tasks. Please remove the handler if the adapter does not react on state changes.

- [ ] Object type hirarchie

  Please note that only ONE object of type 'device' is allowed within the hirarchie. The structure must be
  [device] [- channel ..] - state. 'Device' may be omitted but if present channel must be below device. 'Channel' can be used multiple times. 'States' must not have any children. In addition objects of type 'folder' can be present at any level.

- [ ] structure of jsonConfig is invalid

  i18n is not an object containing translations but a single attribute with boolean avlue (true/false).
  if i18n is true translation data is placed in files located at i18n subtree. You only have to fill in the english version (i18b/en/translations.json and execute `npm run translate` to populate other languages.
  If i18n is set to false (NOT recommended) you may insert the translaations directly inot jsonConfig.json as multilanguage objects just like you did at io.package.json. But note that translation into all supported ioBroker languages is mandatore. You can use https://translator-ui.iobroker.in/ to generate the objects. But using i18b external files is recommended as this enables the usage ob weblate to gain support from external human translators too.

- [ ] jsonCostum.json5 seems to be invalid

  I did not detect handling of custom attributes by your adapter. If I'm correct and did not miss something, please remove jsonCostum.json5. And please remove  common.supprtsCustom from io-package.json too in this case (https://github.com/muffin142/ioBroker.fyta/blob/704eb0f0b6e315e81f4549c693b10308a58547b1/io-package.json#L77)

- [ ] outdated words.js

  words.js seems to be outdated and is empty. JsonConfig does not use this file at all. So please remove the file.

 - [ ] i18n entries must support all languages
 
  When using multilangugae support at objects those entries must contain all supported languages. See https://github.com/muffin142/ioBroker.fyta/blob/704eb0f0b6e315e81f4549c693b10308a58547b1/io-package.json#L140. You can use https://translator-ui.iobroker.in/ to generate i18n blocks.

- [ ] Add timeout to axios calls

  All axios calls should have a timeout set. The default value for axios timeout is 0 which results in no timeout. So an unresponsive remote system could lead to an hanging adapter. Either set an appropiate timeout per axios call or set a default timeout globally for the axios instance.
  
- [ ] consider using adapter.setTimeout / this.setTimeout instead of (standard) setTimeout
  
  The adapter package provides wrapper routines for native setTimeout, setInterval, clearTimeout and clearInterval. Using those routines ensures that timers are cancelled on on load. Additional checks on thomse limits might be performed, too. So consider replacing native setTimeout/clearTimeout by adapter.setTimeout/adapter.clearTimeout or this.setTimeout/this.clearTimeout. The same refers to setInterval/clearInterval.

- [ ] reenable testing
  
  Standard iobroker testing is required but is disabled. Please activate standard ./github/workflows/test-and-release.yml again. Of course its ok to add addional tests but standard github based tests must be active and named as provided by default to enable semi automatic checks. 
    
- [ ] linter setup

  Please use standard iobroker linter setup if acceptable. This is located at @iobroker/eslint-config. See migration guide at https://github.com/ioBroker/ioBroker.eslint-config/blob/main/MIGRATION.md.
    
- [ ] deleting of objects

  Not sure if deleting all objects is really a good idea as this will invalidate all custom configuration (i.e. history settings). But as it's an option set by the user this functionality is acceptable. BUT your code will delte info objects for any instance other than instance 0. This must be fixed. See https://github.com/muffin142/ioBroker.fyta/blob/704eb0f0b6e315e81f4549c693b10308a58547b1/main.js#L47

- [ ] error handling

  - [ ] Stopping processing must be logged as error (and not as info) - see https://github.com/muffin142/ioBroker.fyta/blob/704eb0f0b6e315e81f4549c693b10308a58547b1/main.js#L100 and 
https://github.com/muffin142/ioBroker.fyta/blob/704eb0f0b6e315e81f4549c693b10308a58547b1/main.js#L105
  
  - [ ] Adapter should try to recover from errors at least at limited number of times as accessing an external source may lead to an error. As the interval is set to 30 minutes anyway inlimited retryes would be acceptable too as logging an error every 30 minutes does not overlaod ioBroker for sure. 
    Simply stopping processing becaue i.e. the home router has an interruption due do to ip address change is not userfriendly. Same if the external webpage is ofline for a short time. But IF the adapter stops processing it must terminate itself so that user has a clear indicatio n taht it is no longer working. Simply doing nothing will be interpreted as an error of the adapter by most users.

mcm1957 commented:
@muffin142 

Thnaks for fixing and your work at this adapter.

The following remarks need more attention:

- [ ] Add timeout to axios calls

  All axios calls should have a timeout set. The default value for axios timeout is 0 which results in no timeout. So an unresponsive remote system could lead to an hanging adapter. Either set an appropiate timeout per axios call or set a default timeout globally for the axios instance.

  At least here https://github.com/muffin142/ioBroker.fyta/blob/7f2e0a89f8cc58f3c5a549665e6048ea17f44a9d/main.js#L563 a timeout seems to be missing. (non blocking but strongly recommended)

- [ ] Adapter should try to recover from errors at least at limited number of times as accessing an external source may lead to an error. As the interval is set to 30 minutes anyway inlimited retryes would be acceptable too as logging an error every 30 minutes does not overlaod ioBroker for sure.
Simply stopping processing becaue i.e. the home router has an interruption due do to ip address change is not userfriendly. Same if the external webpage is ofline for a short time. But IF the adapter stops processing it must terminate itself so that user has a clear indicatio n taht it is no longer working. Simply doing nothing will be interpreted as an error of the adapter by most users.

  If I read the code correctly a broken internet connection will stop this adapter. So if your internetconnection is only down a short time at the point the adapter tries to retrieve data the adapter stops. This is an unusual behaviour. Normally adapter simply log an error and retry at the next interval or - especially if the request data at a high ferquency - retry several times before stopping. As this adapter polls only every 30 minutes it would be acceptable that it simply continous to poll after an error. A maximum of 48 errors a day should be acceptable even if the link is permanently down. Please evaluate.

  Feel free to comment if you think stopping is better,

- [ ] Why do you use local translation?

  Using i18n system is more flexible than local hardcoded translations and would enable connection to wablate system. If you really want to keep local translations, the i18n tree is obsolete and should be removed.

**Please add a comment when you have reviewed and fixed the suggestionsor at least commented the suggestions and you think the adapter is ready for a re-review!**

reminder 7.3.2025

mcm1957 commented:
RE-CHECK!

mcm1957 commented:
This adapter has been released to latest repository and should be visible within 24h maximum.

Please create a thread at https://forum.iobroker.net/category/91/tester titled like "Test Adapter " to collect some user feedback and provide a link to this topic when requesting addition to stable repository later.

Note: If an other testing topic already exists, it' OK to continue using this topic too.

If you do not have write access to TESTER Section of forum, please contact HOMORAN at forum using PN/CHAT
